### DISCUSSION TITLE
A question about CWatM output variable: totalPotET

### DISCUSSION THREAD

**Discussion by ray-yicong** [iiasa/CWatM#173] [Q&A] [answered]
I want to ask you a question about CWatM output, recently, I wanted to output varible called totalPotET, but failed, it came to a erros said that :
File "C:\ProgramData\Anaconda3\envs\ycr372\lib\site-packages\numpy\ma\core.py", line 3385, in __setitem__
    _data[indx] = dval
TypeError: NumPy boolean array indexing assignment requires a 0 or 1-dimensional input, input has 2 dimensions

However, I tried other variables related to evapotranspiration are fine, for example：totalET, EWRef.

Can you help me find the problem?

**Comment by mikhailsmilovic** [comment]
Guten Tag @ray-yicong,

totalPotET is a land class-specific variable. That means it can be output as totalPotET[0] for the m/day over forests, totalPotET[1] for the m/day over grasslands, etc. Note then that these are not exactly in units of m, as they are m over the specific land class.

Outputting variables with square brackets do not work with all output types. 
OUT_MAP_Daily and OUT_Map_MonthAvg will work fine.
OUT_Map_AnnualAvg and  "total" maps (ex., MonthTot) will not work, and we must do something else, but perhaps these two output types are sufficient. This "issue" has been included here: https://github.com/iiasa/CWatM/issues/174

**Reply by ray-yicong** [reply]
Sorry i still can't output the totalPotET, i tried both OUT_MAP_Daily and OUT_Map_MonthAvg, the model can run, but it stopped at the second month after the spinup date with the same error: these are the related simulation settings
![image](https://github.com/user-attachments/assets/aa407aab-4418-4c22-bc1a-d4fb8dbf8588)

![image](https://github.com/user-attachments/assets/e4fe97af-fb85-481e-939d-5d326274bc71)
and this is the error
![image](https://github.com/user-attachments/assets/2795e38f-141f-484d-8931-f553d3186bcc)

What is the problem? Can you please help me see again?

**Reply by mikhailsmilovic** [reply, marked as answer]
Guten Morgen, land class-specific variables are to be output indicating the land class, for example
OUT_MAP_Daily = totalPotET[0], totalPotET[1], totalPotET[2], totalPotET[3]
OUT_Map_MonthAvg = totalPotET[0], totalPotET[1], totalPotET[2], totalPotET[3]

Note then that these are not exactly in units of m, as they are m over the specific land class. 
Specifically, m3/(m2 of land class)

0 = forest
1 = grassland
2 = paddy
3 = non-paddy irrigated land

**Reply by ray-yicong** [reply]
Problem solved, it worked！I didn't specify the type of landcover I wanted to output before. That's why the error said that can't has two dimensions.

Thank you again for your detailed answer!

**Comment by ray-yicong** [comment]
Thank you for your detailed answer!




易聪睿
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "iiasa/CWatM"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年12月5日(星期四) 晚上11:38
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [iiasa/CWatM] A question about CWatM output variable: totalPotET (Discussion #173)





 
Guten Tag @ray-yicong,
 
totalPotET is a land class-specific variable. That means it can be output as totalPotET[0] for the m/day over forests, totalPotET[1] for the m/day over grasslands, etc. Note then that these are not exactly in units of m, as they are m over the specific land class.
 
Outputting variables with square brackets do not work with all output types.
 OUT_MAP_Daily and OUT_Map_MonthAvg will work fine.
 OUT_Map_AnnualAvg and  "total" maps (ex., MonthTot) will not work, and we must do something else, but perhaps these two output types are sufficient.
